data.tf

```diff
@@ -89,8 +89,12 @@ EOF
 }
 
 data "template_file" "userdata" {
-  count    = local.worker_group_count
-  template = file("${path.module}/templates/userdata.sh.tpl")
+  count = local.worker_group_count
+  template = lookup(
+    var.worker_groups[count.index],
+    "custom_userdata",
```

Comment:
I love it!

But can we change this key name? Since here we are not supplying the actual user data, we are supplying a template attribute for the `template_file` resource. So how about `userdata_template_file`?

Reply:
Sure.  I'll make that change.